cluster-operator/src/main/java/io/strimzi/operator/cluster/ClusterOperatorConfig.java

```diff
@@ -87,9 +93,25 @@ public static ClusterOperatorConfig fromMap(Map<String, String> map) {
             createClusterRoles = Boolean.parseBoolean(createClusterRolesEnvVar);
         }
 
-        Map<String, String> imageMap = ModelUtils.parseImageMap(map.get(STRIMZI_KAFKA_IMAGE_MAP));
+        KafkaVersion.Lookup lookup = new KafkaVersion.Lookup(
+                ModelUtils.parseImageMap(map.get(STRIMZI_KAFKA_IMAGE_MAP)),
+                ModelUtils.parseImageMap(map.get(STRIMZI_KAFKA_CONNECT_IMAGE_MAP)),
+                ModelUtils.parseImageMap(map.get(STRIMZI_KAFKA_CONNECT_S2I_IMAGE_MAP)),
+                ModelUtils.parseImageMap(map.get(STRIMZI_KAFKA_MIRROR_MAKER_IMAGE_MAP)));
+        for (String version : lookup.supportedVersions()) {
+            if (lookup.kafkaImage(null, version) == null) {
+                LOGGER.warn("{} does not provide an image for version {}", STRIMZI_KAFKA_IMAGE_MAP, version);
+            }
+            if (lookup.kafkaConnectVersion(null, version) == null) {
+                LOGGER.warn("{} does not provide an image for version {}", STRIMZI_KAFKA_CONNECT_IMAGE_MAP, version);
+            }
+            // Need to know whether we're on OS to decide whether to valid s2i
```

Comment:
does this comment mean that there is a missing check on Kafka Connect S2I I guess?

Reply:
That's right, but it's a chicken and egg problem, because we need the config before we've discovered whether we're running on OpenShift. We could factor out some kind of `validate()` method which we called once we knew what we were running on. I don't think this is something I should try to deal with in this PR.

Reply:
ok